services/basic_types.proto

```diff
@@ -142,10 +140,20 @@ message ContractID {
      */
     int64 realmNum = 2;
 
-    /**
-     * A nonnegative number unique within its realm
-     */
-    int64 contractNum = 3;
+    oneof contract {
+        /**
+        * A nonnegative number unique within its realm
+        */
+        int64 contractNum = 3;
+
+        /**
+        * The 20-byte EVM address of the contract to call. Note that if the 
+        * target contract was created via CREATE2, its address will have 
+        * been derived from the <a href="https://eips.ethereum.org/EIPS/eip-1014">EIP-1014</a>
+        * specification, and will not have a simple relation to a 0.0.X id.
```

Comment:
Can we also document the `shard.realm.num` format of the `evm_address`?

Reply:
Added these details.